cql/src/main/java/org/apache/cassandra/stargate/transport/internal/Frame.java

```diff
@@ -191,10 +191,10 @@ Frame decodeFrame(ByteBuf buffer) throws Exception {
         throw ErrorMessage.wrap(e, streamId);
       }
 
-      long bodyLength = buffer.getUnsignedInt(idx);
+      int bodyLength = buffer.getInt(idx);
```

Comment:
The protocol spec is not 100% clear on this, but it says "a 4 byte integer", in any other place if integers are unsigned it's said explicitly. And the rest of the decoder wouldn't work anyway if the length was bigger than an int.